websites/D/dod Games/presence.ts

```diff
@@ -0,0 +1,254 @@
+			);
+			if (!isInGame) {
+				timeStarted = Date.now();
+				isInGame = true;
+			}
+
+			const playerClassNames = [
+					"agonia_player1",
+					"agonia_player2",
+					"agonia_player3",
+					"agonia_player4",
+				],
+				state: { [key: string]: AgoniaPlayerState } = {};
+			for (const playerClassName of playerClassNames) {
+				const player = document.querySelector(`#${playerClassName}`);
+				if (player) {
+					const playerName = player.querySelector(".gnh_name").textContent;
+					if (playerName) {
+						state[playerName] = {
+							score: player.querySelector(".gnh_score_text").textContent,
+							winner: false,
+						};
+					}
+				}
+			}
+
+			const gameOverVisible =
+				document.querySelector("#gameover_content").parentElement.style
+					.display !== "none";
+			if (gameOverVisible) {
+				const gameOverClassNames = [
+					"gameover_user1",
+					"gameover_user2",
+					"gameover_user3",
+					"gameover_user4",
+				];
+				for (const gameOverClassName of gameOverClassNames) {
+					const player = document.querySelector(`#${gameOverClassName}`);
+					if (
+						player &&
+						player.querySelector<HTMLDivElement>(".gameover_userphotowin").style
+							.display !== "none"
+					) {
+						const winnerName = player.getAttribute("shownname");
+						if (winnerName in state) state[winnerName].winner = true;
+					}
+				}
+			}
+			let stateString = "";
+			for (const key in state) {
+				const value = state[key];
+				stateString += `${value.winner ? " 🏆 " : ""}${key}: ${value.score} – `;
+			}
+			if (Object.keys(state).length === nPlayers || gameOverVisible)
+				presenceData.state = stateString.substring(0, stateString.length - 3);
+			else {
+				presenceData.state = `${
+					Object.keys(state).length
+				}/${nPlayers} παίκτες...`;
+				presenceData.buttons = [
+					{
+						label: "Join Game",
+						url: url.textContent,
+					},
+				];
+			}
+			presenceData.details = `Παίζει Αγωνία | 💪🏻 ${document
+				.querySelector(".mytr")
+				.getAttribute("elo")}`;
+			presenceData.startTimestamp = timeStarted;
+			break;
+		} else if (url.textContent.includes("game=tichu&room=")) {
+			// Tichu
+			if (!isInGame) {
+				timeStarted = Date.now();
+				isInGame = true;
+			}
+
+			const teamPlayers: string[] = [],
+				opPlayers: string[] = [],
+				playerState: { [key: string]: TichuPlayerState } = {},
+				bot = document.querySelector(
+					"#nickholder_bottom .playerName"
+				).textContent,
+				up = document.querySelector("#nickholder_up .playerName").textContent,
+				left = document.querySelector(
+					"#nickholder_left .playerName"
+				).textContent,
+				right = document.querySelector(
+					"#nickholder_right .playerName"
+				).textContent;
+
+			if (bot) playerState.bot = { name: bot, bet: null };
+			if (up) playerState.up = { name: up, bet: null };
+			if (left) playerState.left = { name: left, bet: null };
+			if (right) playerState.right = { name: right, bet: null };
```

Comment:
you can probably do the same that you did above with the for loop and the query selectors with template strings